src/main/java/org/lanternpowered/server/script/json/ObjectTypeAdapterFactory.java

```diff
@@ -49,6 +47,7 @@
     private static final String DATA = "data";
 
     private final AbstractObjectTypeRegistryModule registry;
+    // TODO: Might be best to move to a Guava TypeToken in here
```

Comment:
The reason why the gson `TypeToken` is used, is because this is a class that implements a gson library interface and uses that `TypeToken`. We could also provide a constructor with the guava `TypeToken` and convert it to the gson one.